exporters/committime/collector_base.py

```diff
@@ -224,6 +209,72 @@ def get_metric_from_build(self, build, app, namespace, repo_url):
             )
             logging.debug(e, exc_info=True)
             return None
+        except Exception as e:
+            logging.error("Error encountered while getting CommitMetric info:")
+            logging.error(e, exc_info=True)
+            return None
+
+    def _is_metric_ready(self, namespace: str, metric: CommitMetric, build) -> bool:
+        """
+        Determine if a build is ready to be examined.
+
+        There's a few reasons we would stop early:
+          - the build is new/pending/running and doesn't have an image yet.
+          - the build failed/error'd/cancelled.
+        These are valid conditions and we shouldn't clog the logs warning about it.
+        However, if it's new/pending/running and _does_ have an image, we might as well continue.
+        """
+        build_status = get_nested(build, "status.phase", default=None)
+        if build_status in {"Failed", "Error", "Cancelled"}:
+            logging.debug(
+                "Build %s/%s had status %s, skipping",
+                namespace,
+                build.metadata.name,
+                build_status,
+            )
+            return False
+        elif build_status in {"New, Pending", "Running"}:
+            if metric.image_hash is None:
+                logging.debug(
+                    "Build %s/%s has status %s and doesn't have an image_hash yet, skipping",
+                    namespace,
+                    build.metadata.name,
+                    build_status,
+                )
+                return False
+            else:
+                return True
+        else:
+            return True
```

Comment:
So here the logic is as follows:

if something1 goes wrong, return False
if something2 else goes wrong, return False
return True

So if there is something3 that goes wrong you are assuming the metric is ready. I think it would be better to get positive filter rather then negative one, but maybe it's impossible??

Reply:
According to the docs, the only other value it could have is `Complete`. In this case, I'd like to follow the Robustness Principle, and try to process any status we don't recognize. It'll fail if it's missing info anyway.

Reply:
I am fine with this approach. Thanks for clarification